sdk/sdkruntimes/build.go

```diff
@@ -108,27 +109,51 @@ func Build(
 
 		rtName := rtd.Name()
 
+		var (
+			rt   sdkservices.Runtime
+			data *sdkbuildfile.RuntimeData
+		)
+
 		cached := rtCache[rtName.String()]
 		if cached == nil {
-			rt, err := rts.New(ctx, rtName)
-			if err != nil {
+			if rt, err = rts.New(ctx, rtName); err != nil {
 				return nil, fmt.Errorf("new %q: %w", rtName, err)
 			}
 
-			cached = &rtCacheEntry{
-				runtime: rt,
-				data: &sdkbuildfile.RuntimeData{
-					Info: sdkbuildfile.RuntimeInfo{
-						Name: rt.Get().Name(),
-					},
+			data = &sdkbuildfile.RuntimeData{
+				Info: sdkbuildfile.RuntimeInfo{
+					Name: rt.Get().Name(),
 				},
 			}
+		} else {
```

Comment:
I think that instead of this logic, you should check for filewise around line 55 when you walk over the fs. If not - they send the fs to the runtime, otherwise do the loop. Probably in different functions.

The logic here is pretty complex.

Reply:
You're right in principal - problem is that during the walk I when I encounter a dir I don't know if all the files in it are python or not.